androidHyperskillApp/src/main/java/org/hyperskill/app/android/notification/local/DailyStudyReminderLocalNotificationDelegate.kt

```diff
@@ -61,15 +61,24 @@ class DailyStudyReminderLocalNotificationDelegate(
     }
 
     override fun getNextScheduledAt(): Long? {
-        val time = if (wasNotificationTimeSentToServer() || !notificationInteractor.isDailyStudyRemindersEnabled()) {
-            null
-        } else {
+        val time = if (!wasNotificationTimeSentToServer() && notificationInteractor.isDailyStudyRemindersEnabled()) {
             getNextScheduledAtInternal(notificationInteractor.getDailyStudyRemindersIntervalStartHour())
+        } else {
+            null
         }
         Log.i(LOG_TAG, "getNextScheduledAt returns $time")
         return time
     }
 
+    override fun onRescheduleNotificationRequested() {
+        if (!wasNotificationTimeSentToServer() && notificationInteractor.isDailyStudyRemindersEnabled()) {
+            sendNotificationTimeToServer()
+        }
+    }
+
+    override fun shouldRescheduleNotification(): Boolean =
```

Comment:
I suppose that the `shouldRescheduleNotification()` function can be used in both `getNextScheduledAt()` and `onRescheduleNotificationRequested()` to reduce code duplication